packages/cosmos_ui_components/lib/components/cosmos_elevated_button.dart

```diff
@@ -34,14 +34,15 @@ class CosmosElevatedButton extends StatelessWidget {
   Widget build(BuildContext context) {
     final theme = CosmosTheme.of(context);
     return ElevatedButton(
-      onPressed: onTap,
+      onPressed: onTap ?? () {},
```

Comment:
can't we simply update the "disabled" colors for the button? instead of all those  if/elses ?
i.e:
```dart
ElevatedButton(
    style: onTap == null ? _disabledStyle() : _enabledStyle(),
    ...
)
```

Reply:
Yes we can do that, still we would have to provide an empty function to `onTap`, because if `onTap` is `null`, Flutter's internal code automatically applies disabled style and the button disappears on dark mode. Therefore, in case of a `null` `onTap` I had to provide it an empty function and manually remove the splash effect and shadow in order to make it feel like it's not tappable.

Reply:
you still should be able to alternate the way flutter displays disabled button: https://www.flutterbeads.com/elevated-button-color-in-flutter/#Customize-the-Elevated-Button-Color-for-Disable-State